packages/gatsby/src/commands/develop-process.ts

```diff
@@ -209,29 +226,87 @@ module.exports = async (program: IProgram): Promise<void> => {
         },
       },
       doingEverythingElse: {
+        on: {
+          QUERY_FILE_CHANGED,
+          ADD_NODE_MUTATION,
+          WEBHOOK_RECEIVED,
+        },
```

Comment:
I find this pattern a bit confusing because:
 - this doesn't say what those transitions will do
 - https://github.com/gatsbyjs/gatsby/pull/25479/files#diff-91e040b1a5f46cf5babca70ea2ebc333R197 handles `QUERY_FILE_CHANGED` but in different way - so while those are shared in many places - they are not in every place, which is fine, but current way would let me to believe otherwise

Can we instead of exporting `QUERY_FILE_CHANGED` (etc), name those exports in a way that would annotate what it does (for readability).

So for example:

```diff
+/**
+ * Shared event hander for `QUERY_FILE_CHANGED` event that applies in most of the cases
+ */
-export const QUERY_FILE_CHANGED = {
+export const markQueryFilesDirtySideEffect = {
  actions: `markQueryFilesDirty`,
}
```

and

```diff
        on: {
-         QUERY_FILE_CHANGED,
+         QUERY_FILE_CHANGED: markQueryFilesDirtySideEffect,
        },
```

Reply:
Good point. Terseness got the better of me.

Reply:
Other thought is if those are shared, could we use "global transitions" (not sure what they are called in xstate) like in https://xstate.js.org/viz/?gist=610bc765e60b10833e63ade8229860ef

where we have general handler on the machine itself (instead of on states) and then for specific states we have overwrites?

I do still like explicitness better (tho we might end up in situations where action is not handled at all if we forget about handling those in some states - that's easy to miss)

Reply:
Yes, that's the best way. I'll do that.